spec/models/swift_storage_provider_spec.rb

```diff
@@ -7,6 +7,274 @@
   let(:filename) {'text_file.txt'}
   subject { storage_provider }
 
+  describe 'StorageProvider Implementation' do
+    let(:expected_project_id) { SecureRandom.uuid }
+    let(:project) { instance_double("Project") }
+    let(:upload) { FactoryBot.create(:upload, :skip_validation) }
+    let(:expected_meta) {
+      {
+      "content-length" => "#{upload.size}"
+      }
+    }
+    let(:chunk) { FactoryBot.create(:chunk, :skip_validation, upload: upload) }
+
+    it_behaves_like 'A StorageProvider'
+
+    it {
```

Comment:
The `it {}` syntax should really only be used for single line tests and shouldn't contain multiple `expect`s. If you have multiple expectations in an `it` block, it should have a description and be in the do/end form.